internal/notifier/opsgenie.go

```diff
@@ -71,6 +71,9 @@ func (s *Opsgenie) Post(ctx context.Context, event eventv1.Event) error {
 		Details:     event.Metadata,
 	}
 
+	// Add severity to the details map.
+	payload.Details["severity"] = event.Severity
```

Comment:
we have to watch for a `nil`map.

I would do something like

```go
	var details = make(map[string]string)
	if event.Metadata != nil {
		details = event.Metadata
	}
	details["severity"] = event.Severity

	payload := OpsgenieAlert{
		Message:     event.InvolvedObject.Kind + "/" + event.InvolvedObject.Name,
		Description: event.Message,
                Details: details,
	}
```

I would also add a test for this.

Reply:
Added it, not sure how to properly write a test for this though.